File: C/Euler003.c
#include <stdio.h>
int main() {
  long target = 600851475143;
  long last = -1;
  while (target % 2 == 0) {
    target /= 2;
    last = 2;
  }

  for (int i = 3; i <= target; i += 2) {
    if (target == 1) {
      break;
    }

    while (target % i == 0) {
      target /= i;
      last = i;
    }
  }

  printf("%ld\n", last);
}

File: Kotlin/Euler003.kt
package Euler003

fun main() {
  val target = 600851475143
  println(primeFactors(target).last())
}

fun primeFactors(of: Long): List<Long> {
  var factors = mutableListOf<Long>()
  var n = of

  while (n % 2L == 0L) {
    n /= 2L
    factors.add(2L)
  }

  for (i in 3..n step 2) {
      if (n == 1L) { break }

      while (n % i == 0L) {
        n /= i
        factors.add(i)
      }
    }

  return factors
}
